File: backend/Business/Models/ProDTO.cs
using Sotlaora.Business.Entities;

namespace Sotlaora.Business.Models
{
    public class ProDTO
    {
        public int Id { get; set; }
        public string Email { get; set; } = null!;
        public string UserName { get; set; } = null!;
        public Role Role { get; set; }
        public DateTime CreatedAt { get; set; }
        public string? Location { get; set; }
        public bool IsOnline { get; set; }
        public string? ImageRef { get; set; }
        public string? PhoneNumber { get; set; }
        public bool VerifiedIdentity { get; set; }
        public DateTime? LastSeen { get; set; }
        public List<SubcategoryDTO> ProSubcategories { get; set; } = new List<SubcategoryDTO>();
        public List<Order> Orders { get; set; } = new List<Order>();
    }

}

File: frontend/types/ProDTO.ts
import { Gender } from "./UserProfile";
import { SubcategoryDTOWithDesc } from "./Category";
import { SubcategoryDTO } from "./ServicePrices";

export interface ProDTO {
    id: string;
    email: string;
    userName: string;
    role: string;
    createdAt: Date;
    location?: string;
    isOnline: boolean;
    lastSeen?: Date;
    proSubcategories: SubcategoryDTO[];
    orders: any[];
    imageRef?: string;
    phoneNumber?: string;
}

export interface ProPublicProfile extends ProDTO {
    reviews: any[];
    reviewsCount: number;
    rating: number;
    bio: string;
    price?: number;
    completedOrdersCount: number;
}

export interface ProProfileDTO {
    phoneNumber: string;
    city: string;
    dateOfBirth?: string;
    gender: Gender;
    bio: string;
    subcategoryDTOs: SubcategoryDTOWithDesc[];
    totalCount: number;
    filledSubcategoriesCount: number;
}

export interface ProCardDTO {
    id: number;
    proId: number;
    userName: string;
    description: string;
    imageRef: string;
    subcategoriesDTO: SubcategoryDTOWithDesc[];
    location: string;
    price?: number;
    rating: number;
    reviewsCount: number;
    verifiedIdentity: boolean;
}